include/mbedtls/ssl_internal.h

```diff
@@ -157,6 +157,24 @@
 extern "C" {
 #endif
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2) && \
+    defined(MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED)    
+/*
+ * Abstraction for a grid of allowed signature-hash-algorithm pairs.
+ */
+struct mbedtls_ssl_sig_hash_set
```

Comment:
Currently, it's a map, but to fully adhere to the standard, it will need to be a set of signature-hash-pairs at some point, and I don't want to rename the type then. That's why I choose this name (I initially did use `mbedtls_ssl_sig_hash_map`), and tried to document accordingly.